### PR TITLE
Adding authentication_keys option to os_volume EP

### PR DESCRIPTION
**Pull request by petr-balogh**
PR in ansible:
https://github.com/ansible/ansible/pull/40294

### CODE REVIEW

**Review by mwperina — APPROVED** [review]

**Review by mwperina — COMMENTED** [review]
Shouldn't we include in the role updated module once it gets merged to Ansible 2.6, so this is still usable with Ansible 2.5?

**Comment by petr-balogh** [comment]
Thx @mwperina I can include module from my PR in ansible, but last time I did it, it didn't work for us because we ran multiple roles i our playbook, and when ansible once load module from different role, it won't reload it from this one (was our case).  So I don't mind to wait with this PR till https://github.com/ansible/ansible/pull/40294 will be merged and 2.6 is out. So should I Include this module (Still waiting if I will get some comments, so once it will be approved I can update here)?

**Comment by mwperina** [comment]
> Thx @mwperina I can include module from my PR in ansible, but last time I did it, it didn't work for us because we ran multiple roles i our playbook, and when ansible once load module from different role, it won't reload it from this one (was our case). So I don't mind to wait with this PR till ansible/ansible#40294 will be merged and 2.6 is out. So should I Include this module (Still waiting if I will get some comments, so once it will be approved I can update here)?

We can't do anything with this bug, but we need to make sure the works fine with Ansible 2.5 when executed standalone. So please add the module with a name ovirt_external_provider_26.py so we know the module can be removed when we bump ansible version requirement for the role

**Review by tareqalayan — APPROVED** [review]

File: library/README.md
Comment (mwperina) [inline library/README.md]: Please rename to `ovirt_extenal_provider_26.py` here

**Review by mwperina — CHANGES_REQUESTED** [review]

File: library/README.md
Comment (petr-balogh) [reply library/README.md]: done

**Comment by petr-balogh** [comment]
Hi, @mwperina I fixed requested change. Thx

File: roles/oVirt.external-providers/tasks/main.yml
Comment (mwperina) [inline roles/oVirt.external-providers/tasks/main.yml]: You need to replace all invocation of `ovirt_external_provider` inside the role with `ovirt_external_provider_26`, otherwise the updated module won't be used

**Review by mwperina — CHANGES_REQUESTED** [review]

**Comment by petr-balogh** [comment]
@mwperina after discussion with @machacekondra we decided to merge this change after 2.6 will be released because there is change with id in storage domains and this won't work with ansible 2.5 cause of this change. So I removed module from this PR.  Going to verify it with our infra now.. will let you know if it's working ok now later on.

**Comment by petr-balogh** [comment]
Hey, Ansible 2.6 is out. Can this change be merged and can you make new release with this change? Thx

**Comment by mwperina** [comment]
> Hey, Ansible 2.6 is out. Can this change be merged and can you make new release with this change? Thx

We haven't bumped requirement to Ansible 2.6, all oVirt roles still requires 2.5+, so you need to include  ovirt_external_provider from Ansible 2.6, rename it to ovirt_external_provider_26 and update all references to it within the role.

Once we will decide to bump oVirt roles requirement to ansible 2.6 we will remove the module and change source code, but for now we are still supporting ansible 2.5+

**Review by mwperina — APPROVED** [review]

**Comment by petr-balogh** [comment]
Thanks, verified in our infra.

**Comment by petr-balogh** [comment]
Hi @machacekondra  can we merge this one? Thanks

**Review by machacekondra — APPROVED** [review]

**Pull request merged**
